experimental/checkpoint_restore.md

```diff
@@ -0,0 +1,154 @@
+# Docker Checkpoint & Restore
+
+Checkpoint & Restore is a new feature that allows you to freeze a running
+container by checkpointing it, which turns its state into a collection of files
+on disk. Later, the container can be restored from the point it was frozen.
+
+This is accomplished using a tool called [CRIU](http://criu.org), which is an
+external dependency of this feature. A good overview of the history of
+checkpoint and restore in Docker is available in this
+[Kubernetes blog post](http://blog.kubernetes.io/2015/07/how-did-quake-demo-from-dockercon-work.html).
+
```

Comment:
We should clearly define the limitations of current implementation. That no filesystem changes are checkpointed, we don't copy container config etc.
Also on code side, if we can detect some known cases that don't work with criu from container config(almost nothing worked for me when I didn't use `-d` for example) it would be good to fail early and with a descriptive error message.